Languages/IronPython/IronPython/Runtime/Operations/StringOps.cs

```diff
@@ -199,9 +199,10 @@ public static class StringOps {
             if (x == null) {
                 return "None";
             }
-            if (x is string) {
-                // check ascii
-                return CheckAsciiString(context, (string)x);
+
+            string xstr = (x as string);
```

Comment:
I don't think so:
```
Python 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:19:22) [MSC v.1500 32 bit (Intel)] on win32
>>> class S(str): pass
...
>>> a = S('foo')
>>> b = str(a)
>>> a is b
False
>>> c = str(b)
>>> c is b
True
```

If the check in `FastNew` would also include `Extensible<string>` then the first check in the sample code would evaluate to `True`.